homeassistant/components/sense/binary_sensor.py

```diff
@@ -6,40 +6,28 @@
     BinarySensorDeviceClass,
     BinarySensorEntity,
 )
-from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers import entity_registry as er
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
-from .const import (
-    ATTRIBUTION,
-    DOMAIN,
-    MDI_ICONS,
-    SENSE_DATA,
-    SENSE_DEVICE_UPDATE,
-    SENSE_DEVICES_DATA,
-    SENSE_DISCOVERED_DEVICES_DATA,
-)
+from . import SenseConfigEntry
+from .const import ATTRIBUTION, DOMAIN, MDI_ICONS, SENSE_DEVICE_UPDATE
 
 _LOGGER = logging.getLogger(__name__)
 
 
 async def async_setup_entry(
     hass: HomeAssistant,
-    config_entry: ConfigEntry,
+    config_entry: SenseConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the Sense binary sensor."""
-    data = hass.data[DOMAIN][config_entry.entry_id][SENSE_DATA]
-    sense_devices_data = hass.data[DOMAIN][config_entry.entry_id][SENSE_DEVICES_DATA]
-    sense_monitor_id = data.sense_monitor_id
+    sense_monitor_id = config_entry.runtime_data.data.sense_monitor_id
 
-    sense_devices = hass.data[DOMAIN][config_entry.entry_id][
-        SENSE_DISCOVERED_DEVICES_DATA
-    ]
+    sense_devices = config_entry.runtime_data.discovered
     devices = [
-        SenseDevice(sense_devices_data, device, sense_monitor_id)
+        SenseDevice(config_entry.runtime_data.device_data, device, sense_monitor_id)
```

Comment:
Would be slightly faster to store device_data in a local var so it doesn't get looked up every loop

Reply:
I wonder how long ago the migration right below was introduced for the unique ids. I can't tell from the commit history. With that gone, we won't need the list comprehension.